### PR TITLE
Separate security config function

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Divide `NewOptionFromSecret` into 2 functions that return a single value, from a single function that returns 2 values
- Improve code readability
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by aliok** [comment]
cc @pierDipi @matzew

File: control-plane/pkg/reconciler/broker/broker.go [hunk 1]
```diff
@@ -105,11 +105,10 @@ func (r *Reconciler) reconcileKind(ctx context.Context, broker *eventing.Broker)
 
 	logger.Debug("config resolved", zap.Any("config", topicConfig))
 
-	securityOption, secret, err := security.NewOptionFromSecret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig}, r.SecretProviderFunc())
+	secret, err := security.Secret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig}, r.SecretProviderFunc())
 	if err != nil {
 		return fmt.Errorf("failed to create security (auth) option: %w", err)
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/broker/broker.go:110]: Nit: should we change this error message?
```suggestion
		return fmt.Errorf("failed to get secret: %w", err)
```

File: control-plane/pkg/reconciler/broker/broker.go [hunk 2]
```diff
@@ -279,11 +281,21 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		return fmt.Errorf("failed to resolve broker config: %w", err)
 	}
 
-	authProvider := &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig}
-	securityOption, _, err := security.NewOptionFromSecret(ctx, authProvider, r.SecretProviderFunc())
+	secret, err := security.Secret(ctx, &security.MTConfigMapSecretLocator{ConfigMap: brokerConfig}, r.SecretProviderFunc())
 	if err != nil {
 		return fmt.Errorf("failed to create security (auth) option: %w", err)
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/broker/broker.go:286]: Ditto
```suggestion
		return fmt.Errorf("failed to get secret: %w", err)
```

File: control-plane/pkg/reconciler/sink/kafka_sink.go [hunk 3]
```diff
@@ -80,10 +80,21 @@ func (r *Reconciler) reconcileKind(ctx context.Context, ks *eventing.KafkaSink)
 		ks.GetStatus().Annotations = make(map[string]string, 1)
 	}
 
-	securityOption, secret, err := security.NewOptionFromSecret(ctx, &SecretLocator{KafkaSink: ks}, r.SecretProviderFunc())
+	secret, err := security.Secret(ctx, &SecretLocator{KafkaSink: ks}, r.SecretProviderFunc())
 	if err != nil {
 		return fmt.Errorf("failed to create auth option: %w", err)
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/sink/kafka_sink.go:85]: Ditto
```suggestion
		return fmt.Errorf("failed to get secret: %w", err)
```

File: control-plane/pkg/reconciler/sink/kafka_sink.go [hunk 4]
```diff
@@ -262,10 +273,22 @@ func (r *Reconciler) finalizeKind(ctx context.Context, ks *eventing.KafkaSink) e
 	// 	- https://cwiki.apache.org/confluence/display/KAFKA/KIP-286%3A+producer.send%28%29+should+not+block+on+metadata+update
 
 	if ks.GetStatus().Annotations[base.TopicOwnerAnnotation] == ControllerTopicOwner {
-		securityOption, _, err := security.NewOptionFromSecret(ctx, &SecretLocator{KafkaSink: ks}, r.SecretProviderFunc())
+		secret, err := security.Secret(ctx, &SecretLocator{KafkaSink: ks}, r.SecretProviderFunc())
 		if err != nil {
 			return fmt.Errorf("failed to create security (auth) option: %w", err)
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/sink/kafka_sink.go:278]: Ditto
```suggestion
			return fmt.Errorf("failed to get secret: %w", err)
```

**Review by pierDipi — COMMENTED** [review]
Thanks! 

Left a minor comment.

File: control-plane/pkg/reconciler/broker/broker.go (hunk 1 above)
Comment (aliok) [reply control-plane/pkg/reconciler/broker/broker.go:110]: sounds good

**Comment by aliok** [comment]
@pierDipi now you will need to talk to CLA bot!

**Comment by pierDipi** [comment]
It seems to be happy today!

/lgtm
/approve

**Pull request merged**